### PR TITLE
Update Tags For CRT

### PR DESCRIPTION
**Pull request by RyanDerr**
# Overview
Update the `build.yml` to include the commit sha as part of production tags as part of `ICU-15357`.

### CODE REVIEW

File: .github/workflows/build.yml
Comment (RyanDerr) [reply .github/workflows/build.yml]: Good callout @tmessi!

**Comment by moduli** [comment]
@tmessi @johanbrandhorst It looks like the CODEOWNERS files has some rules for workflow files requiring `github-secure-boundary` approval, but it looks like that's just a group for the service account? Do we still need that rule? Releng used to be on that rule, too, but it was removed last year.

**Comment by RyanDerr** [comment]
> @tmessi @johanbrandhorst It looks like the CODEOWNERS files has some rules for workflow files requiring `github-secure-boundary` approval, but it looks like that's just a group for the service account? Do we still need that rule? Releng used to be on that rule, too, but it was removed last year.

Yeah not sure if this rule makes much sense as it exists currently, and will prevent any merges unless the service account approves workflow PRs.

**Comment by johanbrandhorst** [comment]
Yeah, looks like we should remove that group from the CODEOWNERs. I opened https://github.com/hashicorp/boundary/pull/5299 but ran into another error, I will reach out to releng.

**Review by johanbrandhorst — APPROVED** [review]

**Comment by johanbrandhorst** [comment]
I will merge https://github.com/hashicorp/boundary/pull/5299 once it is ready, you may need to rebase to have this PR re-evaluate the CODEOWNERS logic

**Pull request merged**
